org.OpenT2T.Sample.SuperPopular.Helpers/js/index.js

```diff
@@ -0,0 +1,167 @@
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Content-length':  JSON.stringify(options.postData).length
+    }
+    
+    options.method = 'PUT'; //the type of request to make
+   
+    //our HTTPS call 
+    var req = https.request(options, (res) => {
+        res.setEncoding('utf8');
+        res.on('data', (chunk) => 
+        {
+             
+        });
+        res.on('end', () => 
+        {
+             deferred.resolve('The ' + apiField + ' changed to ' + value);
+        });
+        res.on('error', (e) => {
+            deferred.reject(e);
+        }); });
+
+
+    req.on('error', (e) => {
+        console.log('problem with request:');
+        deferred.reject(e);
+    });
+
+    req.write(JSON.stringify(options.postData));
+    req.end();   
+
+    //make our Promise and give it back to the caller
+    return deferred.promise;
+    
+   },
+   
+   getLastReading: function(apiField)
+   {
+    
+    var deferred = q.defer();   // q will help us with returning a promise
+    
+    //the headers to make our call
+    options.headers = 
+    {
+        'Authorization': 'Bearer ' + accessToken,
+        'Content-Type': 'application/json',
+        'Accept': 'application/json', 
+    }
+     
+    //our HTTPS call 
+    var req = https.get(options, (res) => {
+     
+       var body = ' '; //holds the chunks of data
+        res.on('data', (chunk) => 
+        {
+          body += chunk; 
+        });
+        res.on('end', () => 
+        {
+            //parse the JSON response and look for the apiField we want in the JSON body
+             var results = JSON.parse(body.toString());
+             var valueToGet = results.data.last_reading[apiField];
+             deferred.resolve(valueToGet);
+        });
+        res.on('error', (e) => {
+            deferred.reject(e);
+        }); });
+
+    req.on('error', (e) => {
+        console.log('problem with request:');
+        deferred.reject(e);
+    });
+    
+    
+    req.end();   
+    //make our Promise and give it back to the caller
+    return deferred.promise;
+    
+   },
+   
+   getValueOfDesiredState: function(apiField) 
+   {
+       
+    var deferred = q.defer();   // q will help us with returning a promise
+
+    //the headers to make our call
+    options.headers = 
+    {
+        'Authorization': 'Bearer ' + accessToken,
+        'Content-Type': 'application/json',
+        'Accept': 'application/json', 
+    }
+     
+    //our HTTPS call 
+    var req = https.get(options, (res) => {
+
+       var body = ' '; //holds the chunks of data
+        res.on('data', (chunk) => 
+        {
+          body += chunk; 
+        });
+        res.on('end', () => 
+        {
+            //parse the JSON response and look for the apiField we want in the JSON body
+             var results = JSON.parse(body.toString());
+             var valueToGet = results.data.desired_state[apiField];
```

Comment:
Also include this line in the try/catch in case the results did not have the expected structure.